### PR TITLE
Allow to disable jarhell check for tests on 2.2 with `test.jarhell.check`

### PR DESCRIPTION
**Pull request by s1monw**
This relates to #16042 where we agreed on adding an opt-out on 2.2 for test
to disable jarhell checks in the BootstrapForTesting.java - for other branches we
will use different solutions.

### CODE REVIEW

**Comment by s1monw** [comment]
@clintongormley can you take a look?

**Comment by clintongormley** [comment]
LGTM

File: core/src/test/java/org/elasticsearch/bootstrap/BootstrapForTesting.java [hunk 1]
```diff
@@ -87,11 +87,13 @@
...
-            JarHell.checkJarHell();
-        } catch (Exception e) {
-            throw new RuntimeException("found jar hell in test classpath", e);
+        if (systemPropertyAsBoolean("tests.jarhell.check", true)) {
+            // check for jar hell
+            try {
+                JarHell.checkJarHell();
+            } catch (Exception e) {
+                throw new RuntimeException("found jar hell in test classpath", e);
+            }
```
Comment (dadoonet) [inline core/src/test/java/org/elasticsearch/bootstrap/BootstrapForTesting.java]: Should we log that jar hell has been deactivated?

Comment (s1monw) [inline core/src/test/java/org/elasticsearch/bootstrap/BootstrapForTesting.java]: there is no logger here anyway... hot topic :)

Comment (dadoonet) [inline core/src/test/java/org/elasticsearch/bootstrap/BootstrapForTesting.java]: haha! I was pretty sure you were going to answer that :)

**Pull request merged**